### PR TITLE
add flag for stratosphere wateruptake treatment

### PR DESCRIPTION
**Pull request by keziming**
This PR contain two targets:
Firstly, set a name list variable as a flag to turn on/off the treatment for stratospheric H2SO4 water uptake. This treatment is a new future as part of MAM5 but it may contribute to up to 3% of the total computational cost. For production run in general, this function can be turned off. The impact on longterm simulation is minor. For studies related to stratospheric chemistry, this feature may be useful. The default setting is to turn off stratospheric H2SO4 water uptake treatment and treat water uptake in the whole atmosphere same. 

Secondly, set a automatic switch between MAM5 and prescribed cmip6 volcanic treatment. MAM5 is the new feature for the V3 and it treats volcanic emission interactively. If MAM5 is chosen in the configuration, the prescribed cmip6 volcanic treatment can be turned off automatically. This can let the atm_in setting easier. Currently, we need to manually set prescribed volcanic data path to "" to avoid that function activated, which could be an issue for people outside aerosol area.   

I hope this PR can be merged into NGD_v3atm to let model run faster and easier to setup. 
The additional tests are on the way and the change of model output should be minor.

### CODE REVIEW

**Review by wlin7 — APPROVED** [review]
The changes look good.
Will need a cleanup when the codes are to merge to E3SM main repo.

File: components/eam/src/chemistry/utils/modal_aero_calcsize.F90 [hunk 1]
```diff
@@ -75,11 +75,11 @@ module modal_aero_calcsize
 integer, parameter :: npair_csizxf = N_DIAG           !total number of possible diagnostic calls
 
 logical :: do_adjust_allowed                          !flag to turn on/off  aerosol size adjustment process
-#if ( defined MODAL_AERO_5MODE )
-logical :: modal_strat_sulfate_aod = .true.           !kzm !flag to turn on/off stratospheric H2SO4 AOD treatment
-#else
-logical :: modal_strat_sulfate_aod = .false.          !no this setting in MAM4
-#endif
+!#if ( defined MODAL_AERO_5MODE )
+!logical :: modal_strat_sulfate_aod = .false.           !kzm !flag to turn on/off stratospheric H2SO4 AOD treatment
```
Comment (hwangacme) [inline components/eam/src/chemistry/utils/modal_aero_calcsize.F90:79]: Should the logical namelist variables include "_treatment" in the name? or this is a local variable that is different from the one defined for the namelist?

**Review by hwangacme — COMMENTED** [review]
@keziming, is the delete of file pp_chemUCI_linozv3_mam5_resus_mom_soag_tag_cnst_no3_oh2.in part of your change? @mingxuanwupnnl was talking about shorten the name of this file. I am a little confused here.

File: components/eam/src/chemistry/utils/modal_aero_calcsize.F90 (hunk 1 above)
Comment (mingxuanwupnnl) [reply components/eam/src/chemistry/utils/modal_aero_calcsize.F90:79]: I think that file, pp_chemUCI_linozv3_mam5_resus_mom_soag_tag_cnst_no3_oh2.in, is for chemUCI-mam5 only. VBS is not included.

Comment (keziming) [reply components/eam/src/chemistry/utils/modal_aero_calcsize.F90:79]: @hwangacme and @mingxuanwupnnl This chemistry file is for MAM5 only. This is updated file let user can do MAM5-UCI simulations. It was missed in this repository.

**Review by hwangacme — APPROVED** [review]

**Review by mingxuanwupnnl — APPROVED** [review]

**Comment by keziming** [comment]
> 

@mingxuanwu For people outside of aerosol, it is likely to forget 'null' the prescribed_volcaero, when using MAM5. At least this setting automatically turn it off while has no impact on any setting without MAM5. You are right, I can change it to a namelist, such as "volcanic SO2 emission flag" latter.

**Comment by wlin7** [comment]
Merged to NGD_v3atm.

**Pull request merged**